File: frontend/vue-vben-admin/apps/web-antd/src/api/business/customer.ts
import { requestClient } from '#/api/request';

export interface CustomerItem {
  id: string;
  title: string;
  type: string;
  content: string;
  isPublished: number;
  publishedAt: string;
  createdAt: string;
}

export interface CustomerListResult {
  items: CustomerItem[];
  total: number;
}

export interface SaveCustomerParams {
  content: string;
  isPublished: number;
  publishedAt?: null | string;
  title: string;
  type: string;
}

export async function getCustomerListApi(params: Record<string, unknown>) {
  return requestClient.get<CustomerListResult>('/business/customer/list', {
    params,
    responseReturn: 'body',
  });
}

export async function createCustomerApi(data: SaveCustomerParams) {
  return requestClient.post<CustomerItem>('/business/customer', data, {
    responseReturn: 'body',
  });
}

export async function updateCustomerApi(id: string, data: SaveCustomerParams) {
  return requestClient.put<CustomerItem>(`/business/customer/${id}`, data, {
    responseReturn: 'body',
  });
}

export async function deleteCustomerApi(id: string) {
  return requestClient.delete<boolean>(`/business/customer/${id}`, {
    responseReturn: 'body',
  });
}

File: src/MiniAdmin.Domain/Entities/Customer.cs
namespace MiniAdmin.Domain.Entities;

public sealed class Customer
{
    public Guid Id { get; set; }

    public Guid? TenantId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Type { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public int IsPublished { get; set; }

    public DateTimeOffset? PublishedAt { get; set; }

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}
